### ISSUE TITLE
Use * instead of Unpack[...] in messages when targeting Python 3.11 or later

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#16445]
On Python 3.11 and later it seems better to use the native syntax with `*` instead of `Unpack[...]` in various messages. 

This is the current behavior:
```py
x: tuple[int, *tuple[int, ...]]
# note: Revealed type is "tuple[builtins.int, Unpack[builtins.tuple[builtins.int, ...]]]"
reveal_type(x)
```
I'd expect the note to look like this:
```
Revealed type is "tuple[builtins.int, *builtins.tuple[builtins.int, ...]]"
```

**Comment by ilevkivskyi** [comment]
Yeah, we already do this for _error_ messages, see https://github.com/python/mypy/blob/master/mypy/messages.py#L2519-L2520. I just noticed one tricky thing btw: `Unpack` can be also used for `TypedDict` unpacking in `**kwargs`, in that case we need to keep it as `Unpack[...]`.

**Comment by AlexWaygood** [comment]
See also https://github.com/python/cpython/pull/104048 and https://github.com/python/typing_extensions/pull/163, where we changed the runtime repr of `Unpack` to reflect that it could now be used in a PEP-692 context as well as a PEP-646 context

**Comment by ilevkivskyi** [comment]
On latest version we show
```
note: Revealed type is "tuple[int, *tuple[int, ...]]"
```
I think this can be closed now.
